### ISSUE TITLE
Disable texture compression on mesa?

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#2078]
As mentioned in ticket #1488:

Shouldn't be texture compression disabled on mesa by default? Generally S3TC is not available in mesa and you must to install additional package to make it working. And I'm not sure if it works properly even if you have it installed.

Atm. texture compression is disabled by default for intel graphics cards (every operating system). But what about linux mesa drivers for nvidia and amd? I suppose that it will have problems. Is there someone who could check it?

**Comment by deveee** [comment]
I can confirm that texture compression crashes STK with mesa using software rendering.

For some reason I can't make my nvidia working with nouveau driver. But it will most probably cause crash too.

I would totally disable texture compression for mesa driver. @hiker can this be done for 0.9 version?

**Comment by hiker** [comment]
I am happy to change this, but a few question:
you mention that this is for software rendering, which in all likelihood will be too slow to run stk anyway. Can we distinguish software rendering from others? I would also need to check if disabling this as default in the option prevents you from enabling it (which people might want to do).

**Comment by deveee** [comment]
This is not a problem only with software rendering, but with mesa in general (intel, nvidia, amd drivers). Intel has already disabled texture compression in graphics restrictions. But most people use proprietary nvidia/amd drivers and I can't personally check it.

Software rendering usually shows something like this (first line is characteristic):
```
OpenGL vendor string: VMware, Inc.
OpenGL renderer string: Gallium 0.4 on llvmpipe (LLVM 3.6, 256 bits)
OpenGL version string: 3.0 Mesa 10.5.2
OpenGL shading language version string: 1.30
OpenGL context flags: (none)
OpenGL extensions:
```

**Comment by hiker** [comment]
Adding something like:
````
  <card contains="Mesa"     os="linux"  disable="TextureCompressionS3TC"/>
  <card contains="Gallium"  os="linux"  disable="TextureCompressionS3TC"/>
````
to the graphical restriction file should do the trick. Could you just test and commit if it works?

**Comment by deveee** [comment]
I did some more tests:
- for intel graphics card texture compression somehow works (there are some graphical glitches, but it doesn't crash)
- I finally was able to run my nvidia using nouveau drivers (using dri prime, because bumblebee+nouveau doesn't work with my graphics card at all). But when I finally was able to see something other than black screen, it was crashing in every case (with and without texture compression) :P These drivers are not in usable state, at least for my graphics card.

These two lines of course work properly.
